app/src/main/java/com/waminiyi/realestatemanager/features/estatesFilter/EstateFilterFragment.kt

```diff
@@ -0,0 +1,368 @@
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import com.google.android.material.chip.Chip
+import com.waminiyi.realestatemanager.R
+import com.waminiyi.realestatemanager.core.model.data.EstateStatus
+import com.waminiyi.realestatemanager.core.model.data.EstateType
+import com.waminiyi.realestatemanager.core.model.data.PointOfInterest
+import com.waminiyi.realestatemanager.core.model.data.Timeframe
+import com.waminiyi.realestatemanager.core.util.remdispatchers.Dispatcher
+import com.waminiyi.realestatemanager.core.util.remdispatchers.RemDispatchers
+import com.waminiyi.realestatemanager.databinding.FragmentEstateFilterBinding
+import com.waminiyi.realestatemanager.features.extensions.afterTextChanged
+import com.waminiyi.realestatemanager.features.extensions.updateValue
+import com.waminiyi.realestatemanager.features.model.asUiEstateType
+import com.waminiyi.realestatemanager.features.model.asUiPointOfInterest
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+private const val MIN_PHOTO_COUNT_MAX_VALUE = 6
+
+@AndroidEntryPoint
+class EstateFilterFragment : Fragment() {
+
+    private var _binding: FragmentEstateFilterBinding? = null
+    private val binding get() = _binding!!
+    private val filterViewModel: EstateFilterViewModel by viewModels()
+
+    @Dispatcher(RemDispatchers.IO)
+    @Inject
+    lateinit var ioDispatcher: CoroutineDispatcher
+    private lateinit var availableStatesAdapter: ArrayAdapter<String>
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentEstateFilterBinding.inflate(inflater, container, false)
+        availableStatesAdapter =
+            ArrayAdapter(requireContext(), android.R.layout.simple_dropdown_item_1line)
+        setUpCityAutocompleteTextView()
+        viewLifecycleOwner.lifecycleScope.launch {
+            filterViewModel.uiState.collect { uiState ->
+                updateUi(uiState)
+            }
+        }
+
+        binding.buttonApplyFilters.setOnClickListener {
+            filterViewModel.saveFilter()
+        }
+        binding.buttonClearFilter.setOnClickListener {
+            filterViewModel.clearFilter()
+        }
+
+        binding.minPriceTextInputEditText.afterTextChanged { price ->
+            if (price.isBlank()) {
+                filterViewModel.updateMinPrice(null)
+            } else {
+                filterViewModel.updateMinPrice(price.toInt())
+            }
```

Comment:
Evite logique dans fragment
```suggestion
            filterViewModel.updateMinPrice(price.toIntOrNull())
```
fait la meme chose dans le reste de ta vue